### ISSUE TITLE
bug: Back button does not work after switching between tabs

### ISSUE THREAD

**Issue by har0ke** [ionic-team/ionic-framework#28419]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

I have the following routes in my vue project that is using tabs:
```
tab1
   child1 
       child2
tab2
tab3
```

After the navigating through `tab1/` -> `tab1/child1/` -> `tab1/child1/child2/` -> `tab1/child1/` (back button) -> `tab2/` (via tab2 button) -> `tab1/child1/` (via tab1 button), the back button does nothing. Only after clicking the back button it a second time, I get navigated back to `tab1/`

[sr.webm](https://github.com/ionic-team/ionic-framework/assets/16851042/6b451063-363c-47e1-ba39-ea230d9acb85)

### Expected Behavior

After the navigating through `tab1/` -> `tab1/child1/` -> `tab1/child1/child2/` -> `tab1/child1/` (back button) -> `tab2/` (via tab2 button) -> `tab1/child1/` (via tab1 button), the back button should navigate the application to `tab1/`

### Steps to Reproduce

1. Clone the reproduction  repository and serve it to a browser with `ionic serve`
3. Navigate `tab1/` -> `tab1/child1/` -> `tab1/child1/child2/` -> `tab1/child1/` (back button) -> `tab2/` (via tab2 button) -> `tab1/child1/` (via tab1 button)
4. Hit the back button (which does nothing)

### Code Reproduction URL

https://github.com/har0ke/ionic-tabs-example

### Ionic Info

Ionic:

   Ionic CLI       : 7.1.1 (/usr/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/vue 7.5.2

Capacitor:

   Capacitor CLI      : 5.5.1
   @capacitor/android : not installed
   @capacitor/core    : 5.5.1
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.7.4

System:

   NodeJS : v20.8.1 (/usr/bin/node)
   npm    : 10.2.1
   OS     : Linux 6.5

**Comment by liamdebeasi** [comment]
Thanks for the issue. I am going to close this as a duplicate of https://github.com/ionic-team/ionic-framework/issues/25213. The root issue is the same as #25213 since the histories are not being preserved when switching across tabs.

`locationHistory` is internal we do not have public documentation. However, the goal of this is to create stack-based navigation. The `locationHistory` variable is the main linear history (similar to the browser's history). And then `tabsHistory` holds the history for each tab. I believe the problem here is in `tabsHistory` where entries are either being duplicated or removed incorrectly. I hope this clarified things a bit. Let me know if you have any additional questions.
